node/src/chain/fsm.rs

```diff
@@ -118,14 +118,16 @@ impl<N: Network, DB: database::DB, VM: vm::VMExecution> SimpleFSM<N, DB, VM> {
             // Clear up all blacklisted blocks
             self.blacklisted_blocks.write().await.clear();
 
+            let network = self.network.read().await;
+            let nonce = network.public_addr().ip().into();
```

Comment:
Yes
As stated by the commit description, this is needed otherwise a receiving node can discard my request if it already receive the same request by another node (unless the kadcast dupemap expires)

Having an IP solve partially this problem, exactly like the previous implementation did (using a counter), since it's very sporadic for a node A to send the same message to a node B in a timeframe of couple of minutes

Reply:
Anyway, I change the generation of the Nonce

Now it start from a u64 depending on the used IPs and then increments.

It's added to any Payload that support it (like GetMempool or GetBlocks) everytime a direct message (sent_to_peer or send_to_alive_peers) is sent

LMK